geode-core/src/test/java/org/apache/geode/internal/cache/ClusterConfigurationLoaderIntegrationTest.java

```diff
@@ -35,25 +41,100 @@
+  private ClusterConfigurationLoader loader;
 
   @Rule
-  public LocatorStarterRule locator = new LocatorStarterRule().withAutoStart();
-
-  private ClusterConfigurationLoader loader;
+  public LocatorStarterRule locator = new LocatorStarterRule();
 
   @Before
   public void before() {
-    loader = new ClusterConfigurationLoader();
+    loader = Mockito.spy(new ClusterConfigurationLoader());
+  }
+
+  @Test
+  public void requestConfigurationFromLocatorsShouldThrowExceptionWhenClusterConfigurationServiceIsNotEnabled() {
+    locator.withProperty("enable-cluster-configuration", "false").startLocator();
+
+    Set<InternalDistributedMember> locators = new HashSet<>();
+    locators.add((InternalDistributedMember) locator.getLocator().getDistributedSystem()
+        .getDistributedMember());
+
+    assertThatThrownBy(() -> loader.requestConfigurationFromLocators("", locators))
+        .isInstanceOf(FunctionException.class).hasCauseInstanceOf(IllegalStateException.class)
+        .hasMessageContaining("The cluster configuration service is not enabled on this member.");
+  }
+
+  @Test
+  public void requestConfigurationFromLocatorsShouldThrowExceptionAfterTheSixthNullConfigurationResponse() {
+    CustomAnswer customAnswer = new CustomAnswer(10);
+    Mockito.doAnswer(customAnswer).when(loader).requestConfigurationFromOneLocator(Mockito.any(),
+        Mockito.anySet());
+
+    locator.withProperty("enable-cluster-configuration", "true").startLocator();
+    Set<InternalDistributedMember> locators = new HashSet<>();
+    locators.add((InternalDistributedMember) locator.getLocator().getDistributedSystem()
+        .getDistributedMember());
+
+    assertThatThrownBy(() -> loader.requestConfigurationFromLocators("", locators))
+        .isInstanceOf(ClusterConfigurationNotAvailableException.class)
+        .hasMessageContaining("Unable to retrieve cluster configuration from the locator.");
+
+    assertThat(customAnswer.calls).isEqualTo(6);
+  }
+
+
+  @Test
+  public void requestConfigurationFromLocatorsShouldCorrectlyLoadTheClusterConfiguration()
+      throws Exception {
+    locator.withProperty("enable-cluster-configuration", "true").startLocator();
+
+    Set<InternalDistributedMember> locators = new HashSet<>();
+    locators.add((InternalDistributedMember) locator.getLocator().getDistributedSystem()
+        .getDistributedMember());
+
+    ConfigurationResponse response = loader.requestConfigurationFromLocators("", locators);
+    Map<String, Configuration> configurationMap = response.getRequestedConfiguration();
+    assertThat(configurationMap.size()).isEqualTo(1);
+    assertThat(configurationMap.get("cluster")).isNotNull();
   }
 
   @Test
-  public void requestForClusterConfiguration() throws Exception {
+  public void requestConfigurationFromLocatorsShouldCorrectlyLoadTheClusterConfigurationEvenAfterSeveralRetries()
+      throws Exception {
+    int mockLimit = new Random().nextInt(6) + 1;
```

Comment:
I really don't like seeing non-deterministic behavior in a test, beyond that which is inherent to the system.  Is there a reason this can't just be explicitly chosen?

Reply:
No reason at all, I've changed it to use the maximum amount of attempts.